### PR TITLE
fix(kms): fix side effect on whole kms app with refetchInterval okms

### PR DESCRIPTION
**Pull request by vovh**
| Question         | Answer
| ---------------- | ---
| Branch?          | feat/key-management-service-B3
| Bug fix?         | yes
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-15170
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [ ] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/key-management-service/src/data/hooks/useKMSServiceInfos.ts [hunk 1]
```diff
@@ -1,16 +1,15 @@
 import { useQuery } from '@tanstack/react-query';
-import { OKMS } from '@/types/okms.type';
 import { ErrorResponse } from '@/types/api.type';
 import { KMSServiceInfos } from '@/types/okmsService.type';
 import { getServiceInfos } from '../api/okmsService';
 
-export const useKMSServiceInfos = (okms?: OKMS) => {
+export const getKMSServiceInfosQueryKey = (okmsId: string) => {
+  return ['okms/service/infos', okmsId];
+};
```
Comment (pauldkn) [inline packages/manager/apps/key-management-service/src/data/hooks/useKMSServiceInfos.ts:8]: **nitpick:** `(okmsId: string) => ['okms/service/infos', okmsId]`

File: packages/manager/apps/key-management-service/src/data/hooks/useOKMS.ts [hunk 2]
```diff
@@ -36,7 +36,6 @@ export const useOKMSById = (okmsId: string) => {
     ...{
       keepPreviousData: true,
```
Comment (pauldkn) [inline packages/manager/apps/key-management-service/src/data/hooks/useOKMS.ts:37]: I know it has not been modified on this commit but `keepPreviousData` is deprecated with tanstackQuery v5 : [docs](https://tanstack.com/query/latest/docs/framework/react/guides/migrating-to-v5#removed-keeppreviousdata-in-favor-of-placeholderdata-identity-function)

File: packages/manager/apps/key-management-service/src/pages/dashboard/generalInformations/updateName/OkmsNameUpdateModal.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  const { data: okmsServiceInfos, isLoading } = useKMSServiceInfos(okmsId);
+  const navigate = useNavigate();
+  const { updateKmsName, isPending } = useUpdateOkmsName({
+    okmsId,
+    onSuccess: () => {
+      navigate('..');
+    },
+    onError: () => {
+      navigate('..');
+    },
```
Comment (pauldkn) [inline packages/manager/apps/key-management-service/src/pages/dashboard/generalInformations/updateName/OkmsNameUpdateModal.tsx:20]: I don't know the requirements for this ticket, but shouldn't we display the error on the modal to let the user retry ?

**Review by pauldkn — APPROVED** [review]

File: packages/manager/apps/key-management-service/src/pages/dashboard/generalInformations/updateName/OkmsNameUpdateModal.tsx (hunk 3 above)
Comment (chipp972) [reply packages/manager/apps/key-management-service/src/pages/dashboard/generalInformations/updateName/OkmsNameUpdateModal.tsx:20]: And you can provide the error prop to UpdateNameModal that will display it automatically

**Pull request closed without merging**